packages/stitch/tests/typeMergingWithDirectives.test.ts

```diff
@@ -0,0 +1,499 @@
+    // 1. the key for that type will be sent to the resolvers first argument.
+    // 2. an array of keys will sent if the resolver returns a list.
+    //
+    // Note: the subschema can rely on the gateway correctly sending the indicated key and
+    // so it is safe to use a non-validated scalar argument. In the next example, the subschema
+    // will choose to strongly type the `keys` argument, but it is not strictly necessary.
+    typeDefs: `
+      ${typeMergingDirectivesTypeDefs}
+      scalar _Key
+      type Query {
+        me: User
+        _users(keys: [_Key!]!): [User] @merge
+      }
+      type User @base(selectionSet: "{ id }") {
+        id: ID!
+        name: String
+        username: String
+      }
+    `,
+    resolvers: {
+      Query: {
+        me: () => users[0],
+        _users: (_root, { keys }) => keys.map((key: Record<string, any>) => users.find(u => u.id === key.id)),
+      },
+    },
+    schemaTransforms: [typeMergingDirectivesValidator],
+  });
+
+  const inventory = [
+    { upc: '1', inStock: true },
+    { upc: '2', inStock: false },
+    { upc: '3', inStock: true }
+  ];
+
+  const inventorySchema = makeExecutableSchema({
+    // @merge directive will direct the gateway to use the tagged resolver to resolve objects
+    // of that type. By default:
+    // 1. the key for that type will be sent to the resolvers first argument.
+    // 2. an array of keys will sent if the resolver returns a list.
+    //
+    // In this example, the key is constructed by using @base and @computed selection sets.
+    // The @computed directive for a given field instructs the gateway to only add the required
+    // additional selections when the tagged field is included within a query. In addition,
+    // the @computed directive will defer resolution of these fields even when queries originate
+    // within this subschema. The resolver for the mostStockedProduct therefore correctly returns
+    // an object with a `upc` property, but without `price` and `weight`. The gateway will use
+    // the `upc` to retrive the `price` and `weight` from the external services and return to this
+    // service for the `shippingEstimate`. Resolution for @computed fields thereby differs when
+    // querying via the gateway versus when querying the subservice directly.
+    //
+    // Note as well how the merging resolver used by the gateway includes the `price` and `weight`
+    // from the key within its internal representation of a Product even though they are not
+    // by this subschema.
+    //
+    // This example strongly types the ProductKey using an input object. This is optional,
+    // as the subschema can rely on the gateway always sending in the correctly typed data.
+    // It is typed correctly; `upc` is always specified, but `price` and `weight` will only
+    // be included when `shippingEstimate` is included within the query.
+    //
+    typeDefs: `
+      ${typeMergingDirectivesTypeDefs}
```

Comment:
Should these import individually? Seems like I might want to use computed without merge, and vice versa.

Reply:
Good idea!